cpp/src/gandiva/gdv_function_stubs.cc

```diff
@@ -794,6 +795,88 @@ const char* gdv_fn_initcap_utf8(int64_t context, const char* data, int32_t data_
   *out_len = out_idx;
   return out;
 }
+
+GANDIVA_EXPORT
+const char* gdv_fn_mask_first_n(int64_t context, const char* data, int32_t data_len,
+                                int32_t n_to_mask, int32_t* out_len) {
+  if (data_len <= 0) {
+    *out_len = 0;
+    return nullptr;
+  }
+
+  if (n_to_mask < 0) {
+    n_to_mask = n_to_mask * (-1);
+  }
+
+  *out_len = data_len;
+
+  char* out = reinterpret_cast<char*>(gdv_fn_context_arena_malloc(context, *out_len));
+  if (out == nullptr) {
+    gdv_fn_context_set_error_msg(context, "Could not allocate memory for output string");
+    *out_len = 0;
+    return nullptr;
+  }
+
+  // do the masking
+  for (int i = 0; i < data_len; ++i) {
```

Comment:
This loop should only be from i = 0; and i < n_to_mask

After this loop, you should just use memcpy() to copy the remaining bytes

Reply:
Done